test/Parse/ConditionalCompilation/wasm32Target.swift

```diff
@@ -8,3 +8,7 @@ var x = C()
 #endif
 #endif
 var y = x
+
+#if !targetEnvironment(threads)
```

Comment:
I'm still wondering whether `#if canImport(Synchronization)` is a better approach for this?

Reply:
Hmm, I feel `targetEnvironment(threads)` is a very straightforward representation considering `threads` is the *environment* part of target triple and other environments like `simulator` in `arm64-apple-ios-simulator` or `macabi` in `x86_64-apple-ios-macabi`. Also I think it's not desirable situation to recommend people use `canImport` for detecting target, which is beyond the original meaning of `canImport` detecting module importability.

Reply:
I think the confusion is that "threads" is not really an "environment", it's a platform feature. "The iOS simulator" is an environment, as is "Mac Catalyst". `canImport()` does seem closer to me to the right mechanism because the question you're asking is, approximately, "can I import the threading library?"

Reply:
Alternatively, perhaps `_runtime(_multithreaded)`?

Reply:
`-threads` implies not only the ability to import the threading library but also different ABI and CPU features. I agree that "threads" is not an actual environment as others, thus I prefer `_runtime(_multithreaded)` now. WDYT @MaxDesiatov ?

Reply:
FWIW `_runtime(_multithreaded)`, however we precisely spell it, would also be useful on other platforms (e.g. Embedded Swift) where the existence of multithreading is not otherwise knowable. I think for Darwin, Linux, and Windows, it would just need to be set to `true`.

Reply:
> would also be useful on other platforms (e.g. Embedded Swift)

+1

Reply:
IIUC in general we now prefer experimental feature flags to underscored symbols, any preferences on this one specifically?

Reply:
I think the feature is very trivial and it's not difficult to keep compatibility with future evolution. So adding a new "experimental feature" sounds too much for me.

Reply:
> I think for Darwin, Linux, and Windows, it would just need to be set to true.

Agreed, it should just be `true` in all of those cases.

> I'm still wondering whether #if !canImport(Synchronization) would be a better approach for something like this?

I would say not. There's no reason you shouldn't be able to e.g. `import Synchronization` on a system that has no threads; IMO synchronization primitives in that case should probably just do nothing, rather than being unimplemented.